bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/AuditorSelector.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.bookkeeper.meta;
+
+import java.io.IOException;
+import java.util.concurrent.Future;
+import org.apache.bookkeeper.meta.exceptions.MetadataException;
+import org.apache.bookkeeper.net.BookieSocketAddress;
+
+/**
+ * Interface for selecting an auditor.
+ */
+public interface AuditorSelector extends AutoCloseable {
+
+    /**
+     * Selector listener to listen on auditor changes.
+     */
+    interface SelectorListener {
+
+        /**
+         * Trigger when it is selected as a leader.
+         */
+        void onLeaderSelected() throws IOException;
```

Comment:
Better 'election' over 'selection'

Reply:
there is already `AuditorElector`. The selector interface is used by `AuditorElector` for selecting who is the leader. That's why we called it `AuditorSelector` rather than `AuditorElector`. 

and selector is pretty standard in curator : https://curator.apache.org/curator-recipes/leader-election.html